contracts/src/v0.8/dev/ArbitrumValidator.sol

```diff
@@ -311,14 +316,16 @@ contract ArbitrumValidator is TypeAndVersionInterface, AggregatorValidatorInterf
   }
 
   /**
-   * @notice Internal method that approximates the `maxSubmissionCost` (using the L1 gas price feed)
+   * @notice Internal method that approximates the `maxSubmissionCost`
    * @dev On L2 this info is available via `ArbRetryableTx.getSubmissionPrice`.
    * @param calldataSizeInBytes xDomain message size in bytes
    */
   function _approximateMaxSubmissionCost(uint256 calldataSizeInBytes) internal view returns (uint256) {
-    (, int256 l1GasPriceInWei, , , ) = AggregatorV3Interface(s_gasConfig.gasPriceL1FeedAddr).latestRoundData();
-    uint256 l1GasPriceEstimate = uint256(l1GasPriceInWei) * 3; // add 200% buffer (price volatility error margin)
-    return (l1GasPriceEstimate * calldataSizeInBytes) / 256 + l1GasPriceEstimate;
+    return
+      IArbitrumDelayedInbox(CROSS_DOMAIN_MESSENGER).calculateRetryableSubmissionFee(
```

Comment:
To clarify, `CROSS_DOMAIN_MESSENGER` is a proxy contract controlled by the Arbitrum team to facilitate upgrades, so we do not need to have a setter for `CROSS_DOMAIN_MESSENGER`?

If we need to change this address in the future we'd want to have a setter function for it.

Reply:
yes I've confirmed that it's a proxy contract controlled by the Arbitrum team.